public/js/group/create.js

```diff
@@ -0,0 +1,121 @@
+const validateGroup = (data) => {
+  const errors = []
+  const room = parseInt(data.get('room'))
+  const name = data.get('name')
+
+  if (!name) {
+    errors.push('A név kitöltése kötelező')
+  }
+
+  if (typeof room !== 'number' || room < 3 || room > 18) {
+    errors.push('A szint csak 3 és 18 közötti szám lehet')
+  }
+  return errors
+}
+
+const handleResponse = async (res) => {
+  switch (res.status) {
+  case 201:
+    location.href = '/groups'
+    break
+  case 400:
+    const data = await res.json()
+    clearMessages()
+    data.errors.forEach((err) => displayMessage('danger', err.msg))
+    break
+  case 401:
+    displayMessage('danger', UNAUTHORIZED_MESSAGE)
+    break
+  }
+}
+
+const addGroup = (event) => {
+  event.preventDefault()
+  const formData = new FormData(event.target)
+  const errors = validateGroup(formData)
+
+  if (errors.length) {
+    clearMessages()
+    errors.forEach((err) => displayMessage('danger', err))
+  } else {
+    fetch('/groups', { method: 'POST', body: formData })
+      .then(async (res) => await handleResponse(res))
```

Comment:
I might be wrong, but I think you can simply pass async function in promise chain like: ` .then(handleResponse)`

Reply:
You're not wrong, will refactor!